### PR TITLE
Shipping zones management improvements

### PR DESCRIPTION
**Pull request by roadster31**
This PR allow a country to be added to several shipping zones, thus allowing an unlimited number of shipping zones. This is specially useful when shipping modules uses different shipping zone configurations.

Additionally, the shipping zone back-office configuration screen has been improved, to allow
1. update of the name of a shipping zone
2. mass-deletion of countries from a zone.

### CODE REVIEW

**Comment by lovenunu** [comment]
You introduced 3 new issues on Scrutinizer :)

**Comment by roadster31** [comment]
This PR is in progress.

**Comment by lovenunu** [comment]
Just checking as scrutinizer don't send errors on new issues ;)

**Comment by roadster31** [comment]
The PR is now finished.

File: core/lib/Thelia/Config/I18n/en_US.php [hunk 1]
```diff
@@ -18,6 +18,7 @@
     'A currency with code "%name" already exists.' => 'A currency with code "%name" already exists.',
     'A descriptive title' => 'A descriptive title',
     'A message with name "%name" already exists.' => 'A message with name "%name" already exists.',
+    'A name suche as Europe or Overseas' => 'A name suche as Europe or Overseas',
```
Comment (lovenunu) [inline core/lib/Thelia/Config/I18n/en_US.php]: Owww, I think I found a typo. ```suche```

File: core/lib/Thelia/Config/Resources/form.xml [hunk 2]
```diff
@@ -114,9 +114,10 @@
         <form name="thelia.admin.area.create" class="Thelia\Form\Area\AreaCreateForm"/>
         <form name="thelia.admin.area.modification" class="Thelia\Form\Area\AreaModificationForm"/>
         <form name="thelia.admin.area.country" class="Thelia\Form\Area\AreaCountryForm"/>
+        <form name="thelia.admin.area.delete.country" class="Thelia\Form\Area\AreaDeleteCountryForm"/>
         <form name="thelia.admin.area.postage" class="Thelia\Form\Area\AreaPostageForm"/>
 
-        <form name="thelia.shopping_zone_area" class="Thelia\Form\ShippingZone\ShippingZoneAddArea"/>
+        <form name="thelia.shipping_zone_area" class="Thelia\Form\ShippingZone\ShippingZoneAddArea"/>
```
Comment (lovenunu) [inline core/lib/Thelia/Config/Resources/form.xml]: This is BC for modules. I think we should keep both and remove the typo in another version

File: setup/update/sql/2.2.0-alpha1.sql [hunk 3]
```diff
@@ -1,53 +1,81 @@
...
+        on delete cascade,
+    constraint `fk_country_area_country_id`
+        foreign key (`country_id`)
+        references `country` (`id`)
+        on update restrict
+        on delete cascade
+) engine=innodb character set='utf8';
+
+-- Initialize the table with existing data
+INSERT INTO `country_area` (`country_id`, `area_id`, `created_at`, `updated_at`) select `id`, `area_id`, NOW(), NOW() FROM `country` WHERE area_id <> NULL
```
Comment (lunika) [inline setup/update/sql/2.2.0-alpha1.sql]: you change all this file from Upper case to lower case and here you write your query in upper case...

Comment (roadster31) [inline setup/update/sql/2.2.0-alpha1.sql]: Ctrl+Shit+U mess.

**Comment by roadster31** [comment]
Thanks :dancer:

**Pull request merged**